### PR TITLE
feat: add demo mode to instance

### PR DESCRIPTION
**Pull request by djaiss**
This PR adds a demo mode to OfficeLife.

The demo mode is made for an instance that is supposed to showcase how OfficeLife behaves with real data.

To enable the demo mode, we need to edit a new .env variable called `DEMO_MODE`. By default, the demo mode is set to false.

Once the demo mode is enabled:

- a new banner will be added to the header of the application, indicating that the instance runs on the demo mode,

image

- some features are deactivated: importing users, adding new users, deleting the account and deleting employees. For those features, we display the following message:

image

- we also add a new .env variable `ENABLE_SIGNUPS` that activates or deactivates the Signup form on the instance.
- when we are in the demo mode, we also indicate the email and the password that we need to use in order to authenticate to the demo instance.

image

### CODE REVIEW

**Comment by djaiss** [comment]
@asbiin could you review?

**Comment by djaiss** [comment]
@asbiin wait, don't review now. Still have changes to do.

**Comment by djaiss** [comment]
@asbiin that should be good.

File: app/Http/Controllers/Auth/RegisterController.php [hunk 1]
```diff
@@ -27,6 +27,10 @@ public function index()
             return redirect('/home');
         }
 
+        if (! config('officelife.enable_signups')) {
+            return redirect('/login');
```
Comment (asbiin) [inline app/Http/Controllers/Auth/RegisterController.php:31]:
```suggestion
            return redirect()->route('login');
```

**Review by asbiin — CHANGES_REQUESTED** [review]
Nice feature!
You deactivated some actions for demo mode at View level. But you also have to check it in the Controller, or someone would be able to forge a request and run it!

**Comment by djaiss** [comment]
>  You deactivated some actions for demo mode at View level. But you also have to check it in the Controller, or someone would be able to forge a request and run it!

You are right. I'll add those changes.

**Pull request merged**